packages/nx/src/plugins/js/utils/register.ts

```diff
@@ -35,6 +35,59 @@ export const registerTsProject = (
   };
 };
 
+export function getSwcTranspiler(
```

Comment:
Needed to enable `strictNullChecks` otherwise TypeScript was allowing this function to return `null`/`undefined` even when the return type was a function.

Reply:
I hear you, and I agree, but I don't think we should do this via an extra tsconfig, since that won't get checked at build time.

We'd eventually like to enable strict mode for the nx package globally,  but can't just yet. I've removed this part of the changes.

Reply:
Sounds good, and strict mode is a great idea for the entire project.